pkg/controller/infrastructure/infraflow/ensure_utils.go

```diff
@@ -296,3 +299,145 @@ func isUserRouter(config *gcp.InfrastructureConfig) bool {
 func isUserVPC(config *gcp.InfrastructureConfig) bool {
 	return config.Networks.VPC != nil && len(config.Networks.VPC.Name) > 0
 }
+
+// isSimilar return whether to slices are similar using reflect.DeepEqual
+//
+// Similar in this context means they contain the same elements, not necessarily in the same order.
+// In essence, this is set equality using DeepEqual.
+func isSimilar[T any](s1, s2 []T) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for _, v := range s1 {
+		if !slices.ContainsFunc(s2, func(e T) bool { return reflect.DeepEqual(v, e) }) {
+			return false
+		}
+	}
+	return true
+}
+
+// firewallUpdate returns a firewall rule that reflects the differences from 'left' towards 'right'.
+//
+// The intended use for the returned firewall rule is to serve as a minimal patch when updating
+// firewall rules on gcp.
+func firewallUpdate(left, right *compute.Firewall) (*compute.Firewall, error) {
```

Comment:
Wouldn't `current` and `new` be more appropriate instead of `left` and `right`?

Reply:
It previously was `old` and `new`, but the linter complains about using `new` (rightfully so, as it shadows a golang keyword)

Reply:
right, makes sense.
maybe then oldF and newF?
I'm not a big fan of left and right :D

Reply:
well, only if the comment below is not true.